packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -107,16 +107,22 @@ export async function executeIntegrationInstance<
   if (options.enableSchemaValidation === true) {
     process.env.ENABLE_GRAPH_OBJECT_SCHEMA_VALIDATION = 'true';
   }
+  const instanceWithTrimmedConfig = trimStringValues(
+    instance,
+    getMaskedFields(config),
+  );
 
   return timeOperation({
     logger,
     metricName: 'duration-collect',
     operation: () =>
       executeWithContext(
         {
-          instance: trimStringValues(instance, getMaskedFields(config)),
+          instance: instanceWithTrimmedConfig,
           logger,
           executionHistory,
+          executionConfig:
+            config.loadExecutionConfig?.(instanceWithTrimmedConfig) || {},
```

Comment:
I don't mind. I'd love to adopt this in the Jira integration, as well as the Qualys integration, at some point, but will need the async form supported for Jira because it makes API calls to calculate the version of the Jira client to create.